multipatch_analysis/fit_average_first_pulse.py

```diff
@@ -0,0 +1,527 @@
+"""Fit average of first pulses in for voltage and current clamp and place in 
+avg_first_pulse_fit table.
+"""
+
+from neuroanalysis.data import Trace, TraceList
+from multipatch_analysis.database import database as db
+import multipatch_analysis.connection_strength as cs 
+from multipatch_analysis.database.database import TableGroup
+import matplotlib.pyplot as plt
+import numpy as np
+import time
+from neuroanalysis.fitting import fit_psp
+import datetime
+import os
+
+commiting = False  # specifies whether to commit results
+time_before_spike = 10.e-3 #time in seconds before spike to start trace waveforms
+
+class FirstPulseFitTableGroup(TableGroup):
+    """Fits first pulse for each individual sweeps.
+    """
+    schemas = {
+        'avg_first_pulse_fit': [
+             """Contains results of psp_fit on spike aligned, average first pulse PSP for each
+            connection that passed qc in current clamp. During the fit the latency is forced
+            to be the value +/-.5 ms found via fitting all of the pulses (available in the 
+            connection_strength.ic_fit_xoffset). The heavily weighted section (meant to 
+            place more importance of the wave form during the rise time) is shifted to 
+            begin at the latency. Created via fit_average_first_pulse.py. 
+            All units in SI.""",
+            ('pair_id', 'pair.id', 'The ID of the entry in the pair table to which these results apply', {'index': True}),
+
+            # current clamp
+            ('ic_amp', 'float', 'fit amplitude of current clamp average first pulses'),
+            ('ic_latency', 'float', 'fit time elapsed since the time of presynaptic spike (max dv/dt) of current clamp data'),
+            ('ic_rise_time', 'float', 'fit rise time of psp of current clamp data'),
+            ('ic_decay_tau', 'float', 'fit decay of psp of current clamp data'),
+            ('ic_avg_psp_data', 'array', 'array of the data voltage waveform used in fitting; starts 10 ms before pre-synaptic spike'),
+            ('ic_avg_psp_fit', 'array', 'fit array of the best fit voltage waveform starting 10 ms before pre-synaptic spike'),
```

Comment:
Is this necessary, given how easy it should be to generate from the fit parameters?

Reply:
Maybe not but it is convenient so I like it.